elliottlib/errata.py

```diff
@@ -635,26 +635,17 @@ def add_jira_bugs_with_retry(advisory_id: int, bugids: List[str], noop: bool = F
         if noop:
             logger.info('Dry run: Would have attached bugs')
             continue
-        results = add_multi_jira_issues(advisory_id, chunk_of_bugs)
-        for i, result in enumerate(results):
-            if result.status_code != 201:
-                rt = add_jira_issue(advisory_id, chunk_of_bugs[i])
-                if rt.status_code != 201:
-                    errata_error = rt.json()
-                    if 'errors' not in errata_error or len(errata_error) != 1:
-                        raise exceptions.ElliottFatalError(f"attach jira bug {chunk_of_bugs[i]} failed with "
-                                                           f"status={rt.status_code} "
-                                                           f"errmsg={rt.json()}")
-                    if 'idsfixed' not in errata_error['errors'] or len(errata_error['errors']) != 1:
-                        raise exceptions.ElliottFatalError(f"attach jira bug {chunk_of_bugs[i]} failed with "
-                                                           f"status={rt.status_code} "
-                                                           f"errmsg={rt.json()}")
-                    for err in errata_error['errors']['idsfixed']:
-                        if 'The issue is filed already in' not in ' '.join(err):
-                            raise exceptions.ElliottFatalError(f"attach jira bug {chunk_of_bugs[i]} failed with "
-                                                               f"status={rt.status_code} "
-                                                               f"errmsg={rt.json()}")
-        logger.info("All not attached jira bugs attached")
+        try:
+            advisory.addJiraIssues(chunk_of_bugs)
+            advisory.commit()
+        except ErrataException as e:
+            attached_bugs = re.findall("Issue (.*) The issue is filed already in", str(e))
+            if attached_bugs:
+                chunk_of_bugs = [b for b in chunk_of_bugs if b not in [b.upper() for b in attached_bugs]]
```

Comment:
I am not in favor of how the "jira already attached to another advisory" error is handled here.
This is a concept "Garbage in, garbage out". We shouldn't pass already attached jiras to this API at the first place. They should be filtered out before reaching here rather.

Reply:
Therefore, my opinion is completely removing the logic of detecting "jira already attached to another advisory" error here, then add a filter to prevent already attached jiras from being swept.

Reply:
It was handled in https://github.com/openshift/elliott/pull/451, but seems there will have a performance issue

Reply:
ok, then I think i can merge this PR to test it in the next week. If it doesn't work well I will prefer to revert this and revisit https://github.com/openshift/elliott/pull/451